core/src/apps/bitcoin/authorize_coinjoin.py

```diff
@@ -0,0 +1,67 @@
+from micropython import const
+
+from trezor import ui
+from trezor.messages.AuthorizeCoinJoin import AuthorizeCoinJoin
+from trezor.messages.Success import Success
+from trezor.strings import format_amount
+from trezor.ui.text import Text
+
+from apps.base import set_authorization
+from apps.common.confirm import require_confirm, require_hold_to_confirm
+from apps.common.paths import validate_path
+
+from . import addresses
+from .authorization import FEE_PER_ANONYMITY_DECIMALS, CoinJoinAuthorization
+from .common import BIP32_WALLET_DEPTH
+from .keychain import get_keychain_for_coin
+
+if False:
+    from trezor import wire
+
+_MAX_COORDINATOR_LEN = const(18)
+
+
+async def authorize_coinjoin(ctx: wire.Context, msg: AuthorizeCoinJoin,) -> Success:
+    keychain, coin = await get_keychain_for_coin(ctx, msg.coin_name)
+
+    if len(msg.coordinator) > _MAX_COORDINATOR_LEN or any(
+        ord(x) < 32 or ord(x) > 126 for x in msg.coordinator
+    ):
+        raise wire.DataError("Invalid coordinator name.")
+
+    await validate_path(
```

Comment:
Why? This follows the same policy that we apply in all other messages, i.e. require user confirmation for non-standard paths. I don't see a reason to deviate from that here.

Reply:
mostly because our change checks are not robust against short `address_n` prefixes, and while I don't mind that in regular transactions, I don't like it in unattended ones.

that can (and should!) of course be fixed on the side of the individual path checks, but OTOH coinjoin is a completely new thing (so not breaking anyone's existing impl) and I don't think it's wrong to have a strong path check that can be weakened later